web/webkit/src/test/scala/net/liftweb/http/SecurityRulesSpec.scala

```diff
@@ -0,0 +1,238 @@
+  "HttpsRules" should {
+    "default to no required HTTPS" in {
+      HttpsRules().headers must be empty
+    }
+
+    "provide a secure variant with 1-year max age and including sub-domains" in {
+      HttpsRules.secure.requiredTime must_== Some(Duration(365, DAYS))
+      HttpsRules.secure.includeSubDomains must_== true
+    }
+
+    "generate a correct Strict-Transport-Security header without sub-domains" in {
+      HttpsRules(Some(Duration(1440, SECONDS)), false).headers must_== List(
+        "Strict-Transport-Security" -> "max-age=1440"
+      )
+    }
+
+    "generate a correct Strict-Transport-Security header including sub-domains" in {
+      HttpsRules(Some(Duration(1440, SECONDS)), true).headers must_== List(
+        "Strict-Transport-Security" -> "max-age=1440 ; includeSubDomains"
+      )
+    }
+  }
+}
+
+class ContentSecurityPolicySpec extends Specification {
+  "ContentSecurityPolicy" should {
+    "default to accepting images from everywhere" in {
+      ContentSecurityPolicy().imageSources must_== List(ContentSourceRestriction.All)
+    }
+
+    "default to allowing script eval and script sources only from self" in {
+      ContentSecurityPolicy().scriptSources must_== List(
+        ContentSourceRestriction.UnsafeEval,
+        ContentSourceRestriction.Self
+      )
+    }
+
+    "default to allowing everything else only from self" in {
+      ContentSecurityPolicy().defaultSources must_== List(ContentSourceRestriction.Self)
+      ContentSecurityPolicy().connectSources must_== Nil
+      ContentSecurityPolicy().fontSources must_== Nil
+      ContentSecurityPolicy().frameSources must_== Nil
+      ContentSecurityPolicy().mediaSources must_== Nil
+      ContentSecurityPolicy().objectSources must_== Nil
+      ContentSecurityPolicy().styleSources must_== Nil
+    }
+
+    "provide a secure setting that drops image sources to the default restrictions" in {
+      ContentSecurityPolicy.secure.defaultSources must_== List(ContentSourceRestriction.Self)
+      ContentSecurityPolicy.secure.imageSources must_== Nil
+      ContentSecurityPolicy.secure.connectSources must_== Nil
+      ContentSecurityPolicy.secure.fontSources must_== Nil
+      ContentSecurityPolicy.secure.frameSources must_== Nil
+      ContentSecurityPolicy.secure.mediaSources must_== Nil
+      ContentSecurityPolicy.secure.objectSources must_== Nil
+      ContentSecurityPolicy.secure.styleSources must_== Nil
+    }
+
+    "default to reporting to the CSP default report URI" in {
+      ContentSecurityPolicy().reportUri must_== Some(ContentSecurityPolicy.defaultReportUri)
+    }
+
+    "provide [X-]Content-Security-Policy if enforcement is enabled" in {
+      ContentSecurityPolicy()
+        .headers(enforce = true, logViolations = true)
+        .collect {
+          case (headerName, _) if headerName.contains("Content-Security-Policy") =>
+            headerName
+        } must_== List("Content-Security-Policy", "X-Content-Security-Policy")
```

Comment:
Because this is security stuff, I'm trying to be a little more specific about checking that it includes these and only these values and headers. Maybe there's a verified that checks that all contents are exactly the same but not necessarily in order (`must contain` doesn't fail if there are unexpected things, iirc). 

That said, the error messages aren't particularly hard to parse imo: you get both the result and the expectation printed out (plus that weird string diff, but that's not useful 99% of the time anyway), right?